spaceship/lib/spaceship/portal/device.rb

```diff
@@ -108,31 +116,57 @@ def all_for_profile_type(profile_type)
           end
         end
 
+        # <b>DEPRECATED:</b> Use <tt>find_by_platform</tt> instead.
         # @param mac [Bool] Searches for Macs if true
         # @return (Device) Find a device based on the ID of the device. *Attention*:
         #  This is *not* the UDID. nil if no device was found.
         def find(device_id, mac: false)
-          all(mac: mac).find do |device|
+          puts '`find` is deprecated. Please use `find_by_platform` instead.'.red
+          find_by_platform(device_id, platform: mac ? Spaceship::Portal::App::MAC : Spaceship::Portal::App::IOS)
+        end
+
+        # @param platform [String] Searches for specified platform
+        # @return (Device) Find a device based on the ID of the device. *Attention*:
+        #  This is *not* the UDID. nil if no device was found.
+        def find_by_platform(device_id, platform: Spaceship::Portal::App::IOS)
+          all_by_platform(platform: platform).detect do |device|
             device.id == device_id
           end
         end
 
+        # <b>DEPRECATED:</b> Use <tt>find_by_udid_by_platform</tt> instead.
         # @param mac [Bool] Searches for Macs if true
         # @return (Device) Find a device based on the UDID of the device. nil if no device was found.
         def find_by_udid(device_udid, mac: false)
-          all(mac: mac).find do |device|
+          puts '`find_by_udid` is deprecated. Please use `find_by_udid_by_platform` instead.'.red
+          find_by_udid_by_platform(device_udid, platform: mac ? Spaceship::Portal::App::MAC : Spaceship::Portal::App::IOS)
+        end
+
+        # @param platform [String] Searches for specified platform
+        # @return (Device) Find a device based on the UDID of the device. nil if no device was found.
+        def find_by_udid_by_platform(device_udid, platform: Spaceship::Portal::App::IOS)
+          all_by_platform(platform: platform).detect do |device|
             device.udid.casecmp(device_udid) == 0
           end
         end
 
+        # <b>DEPRECATED:</b> Use <tt>find_by_name_by_platform</tt> instead.
         # @param mac [Bool] Searches for Macs if true
         # @return (Device) Find a device based on its name. nil if no device was found.
         def find_by_name(device_name, mac: false)
-          all(mac: mac).find do |device|
+          puts '`find_by_name` is deprecated. Please use `find_by_name_by_platform` instead.'.red
```

Comment:
`spaceship` shouldn't use any colored output, please remove all `.red` and similar from this PR.